sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InMemoryWatermarkManager.java

```diff
@@ -1297,7 +1308,9 @@ public String toString() {
   private static class WindowedValueByTimestampComparator extends Ordering<WindowedValue<?>> {
     @Override
     public int compare(WindowedValue<?> o1, WindowedValue<?> o2) {
-      return o1.getTimestamp().compareTo(o2.getTimestamp());
+      return ComparisonChain.start()
+          .compare(o1.getTimestamp(), o2.getTimestamp())
```

Comment:
The ordering of the windows is not meaningful in this context (and windows aren't generally comparable), so I don't think so.